processing/src/main/java/io/druid/segment/data/GenericIndexed.java

```diff
@@ -407,35 +216,56 @@ public T get(int index)
   @Override
   public int indexOf(T value)
   {
-    return bufferIndexed.indexOf(value);
+    return indexOf(this, value);
+  }
+
+  private int indexOf(Indexed<T> indexed, T value)
```

Comment:
It's very hard to construct such benchmark, so that you actually test something meaningful. If you just create large GenericIndexed before the main benchmark run, it could happen before C1/C2 inlining and the actual benchmark (v1), when code is inlined, won't "account" v2 at all. If you ensure C1/C2 compiler is run with v2, it will make v2-optimized assembly, definitely not something you want to test. Also there could be re-compilations during the "main" run..

To benchmark something, we should mimic the real production load, but it doesn't even exist yet.

https://shipilev.net/blog/2015/black-magic-method-dispatch/ tells that boolean flag is at least not worse than "virtual call" in any situation. Often it's better.

GenericIndexed is only relatively hot, only `get()` is called during queries more than "several" times, and nearly not every query, only specific types of queries. Also `get()` performance is dominated by the performance of object deserialization, which is very suboptimal now and nobody is fighting to death trying to optimize it (e. g. reading UTF-8).

Given all this I'm not willing to spend a whole day constructing a proper v1+v2 benchmark